Cython/Debugger/Cygdb.py

```diff
@@ -89,9 +89,7 @@ def make_command_file(path_to_debug_info, prefix_code='',
                 f.write(textwrap.dedent('''\
                     python
                     import sys
-                    try:
-                        gdb.lookup_type('PyModuleObject')
-                    except RuntimeError:
+                    if not hasattr(gdb.selected_inferior().progspace, "symbol_file"):
```

Comment:
This seems worth a comment. (Also for the other occurrence.)

Reply:
In my testing environment, this part of the symbol table hadn't been loaded by the time this was executed. For reference, [`PyModuleObject` is defined here](https://github.com/python/cpython/blob/fd49e226700e2483a452c3c92da6f15d822ae054/Include/internal/pycore_moduleobject.h#L25). Not sure if it got moved since this was written or GDB started loading symbol tables more lazily. Regardless, it seems better practice to check that GDB has a symbol table corresponding to the executable rather than checking for the existence of a specific type, even if it is in the core definitions.

Does the original setup work on your setup? Was there something more specific you wanted a comment about? Here's documentation on [inferiors and progspaces](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Inferiors-Connections-and-Programs.html) and on [the `symbol_file` attribute specifically](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Progspaces-In-Python.html#index-Progspace_002esymbol_005ffile)

Reply:
Then it's just a matter of wording. The words used in the expression are not very obvious (to me).
```suggestion
                    # Check if the Python executable provides a symbol table.
                    if not hasattr(gdb.selected_inferior().progspace, "symbol_file"):
```